tests/Nox.Types.Tests/Types/User/UserTests.cs

```diff
@@ -0,0 +1,154 @@
+﻿using FluentAssertions;
+using Nox.Enums;
+using Nox.TypeOptions;
+
+namespace Nox.Types.Tests.Types;
+
+public class UserTests
+{
+    [Fact]
+    public void User_Constructor_Guid_ReturnsSameValue()
+    {
+        var userGuidString = Guid.NewGuid().ToString();
+
+        var user = User.From(userGuidString);
+
+        user.Value.Should().Be(userGuidString);
+    }
+
+    [Fact]
+    public void User_Constructor_Email_ReturnsSameValue()
+    {
+        var userEmailString = "user@iwgplc.com";
+
+        var user = User.From(userEmailString, new UserTypeOptions { UserFormatType = UserFormatType.Email});
+
+        user.Value.Should().Be(userEmailString);
+    }
+
+    [Fact]
+    public void UserTypeOptions_Constructor_ReturnsDefaults()
+    {
+        var userTypeOptions = new UserTypeOptions();
+
+        userTypeOptions.MinLength = 1;
+        userTypeOptions.MaxLength = 511;
+        userTypeOptions.UserFormatType = UserFormatType.Guid;
+    }
+
+    [Fact]
+    public void User_Constructor_SpecifyingMaxLength_WithLongerLengthInput_ThrowsValidationException()
+    {
+        Assert.Throws<TypeValidationException>(() => _ =
```

Comment:
Can we assert the exceptions using FluentAssertions?
We should assert that the exception has been thrown and that it contains expected errors.